crates/wasi-nn/src/backend/openvino.rs

```diff
@@ -157,24 +148,51 @@ impl From<SetupError> for BackendError {
 
 /// Return the execution target string expected by OpenVINO from the
 /// `ExecutionTarget` enum provided by wasi-nn.
-fn map_execution_target_to_string(target: ExecutionTarget) -> &'static str {
+fn map_execution_target_to_string(target: ExecutionTarget) -> DeviceType<'static> {
     match target {
-        ExecutionTarget::Cpu => "CPU",
-        ExecutionTarget::Gpu => "GPU",
+        ExecutionTarget::Cpu => DeviceType::CPU,
+        ExecutionTarget::Gpu => DeviceType::GPU,
         ExecutionTarget::Tpu => unimplemented!("OpenVINO does not support TPU execution targets"),
     }
 }
 
 /// Return OpenVINO's precision type for the `TensorType` enum provided by
 /// wasi-nn.
-fn map_tensor_type_to_precision(tensor_type: TensorType) -> openvino::Precision {
+fn map_tensor_type_to_precision(tensor_type: TensorType) -> openvino::ElementType {
     match tensor_type {
-        TensorType::Fp16 => Precision::FP16,
-        TensorType::Fp32 => Precision::FP32,
-        TensorType::Fp64 => Precision::FP64,
-        TensorType::U8 => Precision::U8,
-        TensorType::I32 => Precision::I32,
-        TensorType::I64 => Precision::I64,
-        TensorType::Bf16 => todo!("not yet supported in `openvino` bindings"),
+        TensorType::Fp16 => ElementType::F16,
+        TensorType::Fp32 => ElementType::F32,
+        TensorType::Fp64 => ElementType::F64,
+        TensorType::U8 => ElementType::U8,
+        TensorType::I32 => ElementType::I32,
+        TensorType::I64 => ElementType::I64,
+        TensorType::Bf16 => ElementType::Bf16,
+    }
+}
+
+/// Return the `TensorType` enum provided by wasi-nn for OpenVINO's precision type
+fn map_precision_to_tensor_type(element_type: openvino::ElementType) -> TensorType {
+    match element_type {
+        ElementType::F16 => TensorType::Fp16,
+        ElementType::F32 => TensorType::Fp32,
+        ElementType::F64 => TensorType::Fp64,
+        ElementType::U8 => TensorType::U8,
+        ElementType::I32 => TensorType::I32,
+        ElementType::I64 => TensorType::I64,
+        ElementType::Bf16 => TensorType::Bf16,
+        ElementType::Undefined => todo!("not yet supported in wasi-nn `TensorType`"),
```

Comment:
I think this could get cleaned up to something like: `ElementType::Undefined | ElementType::Dynamic | ... => Err(BackendError::SomeNewVariant("{:?} is not yet supported in the openvino backend", element_type))`